src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1

```diff
@@ -2955,99 +2955,107 @@ class Project: ADOSVTBase
+            $inputbody.dataProviderContext.properties.permissionSetToken = "repoV2/$($permissionSetToken)"
 
-                    $accessList += $responseObj.dataProviders."ms.vss-admin-web.security-view-members-data-provider".identities | Where-Object { $_.subjectKind -eq "group" } | ForEach-Object {
-                        $identity = $_
-                        $body.dataProviderContext.properties.accountName = $_.principalName
-                        $body.dataProviderContext.properties.subjectDescriptor = $_.descriptor
+            # Checking if Inherited permissions are allowed or not.
+            $allowPermissionBits = @(1)
+            if ([Helpers]::CheckMember($this.ControlSettings, "Repo.CheckForInheritedPermissions") -and $this.ControlSettings.Repo.CheckForInheritedPermissions) {
+                #allow permission bit for inherited permission is '3'
+                $allowPermissionBits = @(1,3)
+            }
 
-                        $identityPermissions = [WebRequestHelper]::InvokePostWebRequest($url, $body);
-                        $configuredPermissions = $identityPermissions.dataproviders."ms.vss-admin-web.security-view-permissions-data-provider".subjectPermissions | Where-Object {$_.permissionDisplayString -ne 'Not set'}
-                        return @{ IdentityName = $identity.DisplayName; IdentityType = $identity.subjectKind; Permissions = ($configuredPermissions | Select-Object @{Name="Name"; Expression = {$_.displayName}},@{Name="Permission"; Expression = {$_.permissionDisplayString}}) }
-                    }
+            # Get list of all users and groups granted permissions on all repositories
+            $responseObj = [WebRequestHelper]::InvokePostWebRequest($url, $inputbody);
 
-                    $accessList += $responseObj.dataProviders."ms.vss-admin-web.security-view-members-data-provider".identities | Where-Object { $_.subjectKind -eq "user" } | ForEach-Object {
-                        $identity = $_
-                        $body.dataProviderContext.properties.subjectDescriptor = $_.descriptor
+            # Iterate through each user/group to fetch detailed permissions list
+            if([Helpers]::CheckMember($responseObj[0],"dataProviders") -and ($responseObj[0].dataProviders.'ms.vss-admin-web.security-view-members-data-provider') -and ([Helpers]::CheckMember($responseObj[0].dataProviders.'ms.vss-admin-web.security-view-members-data-provider',"identities")))
+            {
+                $body = '{"contributionIds":["ms.vss-admin-web.security-view-permissions-data-provider"],"dataProviderContext":{"properties":{"subjectDescriptor":"","permissionSetId": "2e9eb7ed-3c0a-47d4-87c1-0ffdd275fd87","permissionSetToken":"","accountName":"","sourcePage":{"url":"","routeId":"ms.vss-admin-web.project-admin-hub-route","routeValues":{"project":"","adminPivot":"repositories","controller":"ContributedPage","action":"Execute"}}}}}' | ConvertFrom-Json
```

Comment:
here we are using the constant permission set Id, so please make it consistent by adding that constant in above places as well